File: CISC 131 py/tiling.py
def tiling():
    print("What size board would you like? (Width) ", end="")
    width = int(input())
    print("What size board would you like? (Height) ", end="")
    height = int(input())
    print("What pattern would you like (1-3): ", end="")
    pattern = int(input())
    board = []
    if pattern == 1:
        for row in range(height):
            board.append([])
            for column in range(width):
                if (row + column)%2 == 0:
                    board[row].append('O')
                else:
                    board[row].append('X')
    elif pattern == 2:
        for row in range(height):
            board.append([])
            for column in range(width):
                if (row + column)%3 == 0:
                    board[row].append('O')
                else:
                    board[row].append('X')
    elif pattern == 3:
        for row in range(height):
            board.append([])
            for column in range(width):
                if column % 2 == 0:
                    board[row].append('O')
                else:
                    board[row].append('X')
    printBoard(board)

def printBoard(board):
    for i in range(len(board)):
        for j in range(len(board[i])):
            print(board[i][j], end="")
        print()

tiling()

File: CISC 131 c/tiling.c
#include <stdio.h>
int main(){
    int l = 0;
    int w = 0; 
    printf("What size LxW board would you like? (Length): ");
    scanf(" %i", &l);
    printf("What size LxW board would you like? (Width): ");
    scanf(" %i", &w);
    char board[l][w];
    /** Fill in the board **/
    int i = 0;
    int j = 0;
    while(i < l) /** Rows **/
    {
         j = 0;
        while(j < w) /** Cols **/
        {
            if( (i+j)%2 == 0) /** Even **/
            {
                 board[i][j] = 'O';
            }
            else /** Odd **/
            {
                board[i][j] = 'X';
            }
            j++; 
        } 
        i++;
    } 
    /** ADD YOUR PRINT CODE HERE **/
    i = 0;
    while (i<l){
        j = 0;
        while (j<w){
            printf("%c", board[i][j]);
            j++;
        }
        i++;
        printf("\n");
    }
    return 0; 
} 